lib/lvol/lvol.c

```diff
@@ -1213,6 +1213,12 @@ create_lvol_snapshot(struct spdk_lvol *origlvol, const char *snapshot_name,
 	}
 
 	origblob = origlvol->blob;
+	/*
+	 * Reset used clusters cache if blob is thin provisioned and new snapshot
+	 *  is created from it.
+	 */
+	spdk_blob_reset_used_clusters_cache(origblob);
```

Comment:
We can move it little down after basic snapshot validation is done. 
When snapshot blobstore creation start, We work on new lvol and blobstore. Post that getting back the original lvol and blob to operate is little tricky and may not be very clear to walkthrough.

Other option to handle it in dataplane code by calling SPDK API of reset.

Reply:
SPDK callback function takes new lvol and blob as argument, hence it can't be called there.
We can call this new function in dataplane callback function of snapshot creation.